backend/src/data/repositories/tenant/tenant.repository.ts

```diff
@@ -26,6 +26,17 @@ class Tenant {
     return Tenant.modelToEntity(tenant);
   }
 
+  async updateById(id: string, name: string): Promise<TenantEntity | null> {
```

Comment:
Repositories should not contain `updateById`-like methods. Here is a quick explanation:
Often when we need to update something we also decide to have some validation before updating. After some time these validations overwhelm repository methods and make them unsopportable.
If you want update data you need to create update methods on Entities and add `save(tenant: TenantEntity)` method to repository